### DISCUSSION TITLE
How to determine which is the best cluster marker from FindMarkers() outputs?

### DISCUSSION THREAD

**Discussion by denvercal1234GitHub** [satijalab/seurat#4476] [Q&A] [answered]
Following up on #3817 and #736, I have 10 clusters. I wonder: 

(1) Whether the 2 lines of code below are exactly the same in terms of the outputs? 

- FindMarkers() setting ident.1 = 1 and ident.2 = NULL 
- FindMarkers() setting ident.1 = cluster 1 and ident.2 = c(0, 2,3,4,5,6,7,8,9,10)

(2) When I do either the code, the number of differentially expressed genes defining cluster 1 from all 9 other clusters is always around 700-800 (including negative markers). This makes it very hard to pick which gene markers to study further that truly distinguish (define) the cluster 1. 

Is it sensible to do FindMarkers() setting ident.1 = 1, ident.2 = 0, then ident.1=1, ident.2=2, and so on then find intersection of the lists of DE genes? But this way, there is no multiple comparison correction... 

Anyone has a better idea of how to narrow down this list of DE genes? 

Thank you for your help!

**Comment by timoast** [comment, marked as answer]
1. If you have 11 clusters, then yes they are the same (comparing cluster 1 vs all other cells)
2. To select "best" markers, you can filter based on p-value, fold-change, and the specificity for a cell type (whether the gene was also DE in another cell type)
